packages/hasher/src/PackageTree.ts

```diff
@@ -110,7 +110,10 @@ export class PackageTree {
 
   getPackageFiles(packageName: string, patterns: string[]) {
     const { root, packageInfos } = this.options;
-    const packagePath = path.relative(root, path.dirname(packageInfos[packageName].packageJsonPath)).replace(/\\/g, "/");
+    // If packageName is given, look up the directory if its package.json.  Otherwise, return files that are not in any package.
```

Comment:
In what case would `packageName` be empty? Can you update the PR with a description of the scenario this is trying to fix?

Reply:
If you are referencing files in the workspace root, the package name is empty.  For example if you specify `#package.json` or `#scripts/lage/babel-worker.cjs`.

I'll update the comment to be clearer.